schema_utils_test.py

```diff
@@ -987,3 +988,16 @@ def test_normalizer_sanitize_url(self):
             'Invalid URL: Sanitized URL should start with \'http://\' or'
             ' \'https://\'; received www.oppia.org'):
             sanitize_url('www.oppia.org')
+
+    def test_is_valid_ratio_input_validator(self):
```

Comment:
Please add lots more tests -- empty string, single number, multi-digit numbers, letters, etc. 

There should probably be at least 3 times as many cases as you currently have.

Reply:
Okay would do that

Reply:
Also: spaces in various places, starting with colon, using semicolons instead of colons (typo), decimals, fractions representing ratios, digits starting with zero, and maybe negatives & invalid ratios.

@seanlip should we be considering '0:0' a valid ratio? I can see merit in supporting 'x:0' and '0:x'. Also, should we be supporting negative ratios? My inclination is that '0:0' is always incorrect, and that negative ratios should be conditionally supported based on a configuration value.

Reply:
Ah good point, I missed this. For now, let's require that elements in a ratio should always be positive whole numbers (not zero, not negative -- we never use those cases).